### ISSUE TITLE
Cannot quantize YOLOv26 model using model.export()

### ISSUE THREAD

**Issue by kjmiller944** [ultralytics/ultralytics#24633]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

I am trying to quantize a yolov26m model using the following code.

```
from ultralytics import YOLO

model = YOLO("store/models/yolo26m.pt")

results = model.export(
    format="tflite",
    imgsz=640,
    int8=True,
    data="visdrone-2019-det-optimal-split_small/dataset.yaml",
    verbose=True,
    fraction=0.30
)
```
 It is hanging for hours at the following stage and never completes building the tflite file.

```
W0000 00:00:1779979615.732486 4608019 tf_tfl_flatbuffer_helpers.cc:365] Ignored output_format.
W0000 00:00:1779979615.732534 4608019 tf_tfl_flatbuffer_helpers.cc:368] Ignored drop_control_dependency.
I0000 00:00:1779979615.785253 4608019 mlir_graph_optimization_pass.cc:425] MLIR V1 optimization pass is not enabled
```

### Environment

Ultralytics 8.4.56 🚀 Python-3.12.13 torch-2.12.0 CPU (Apple M2 Max)
Setup complete ✅ (12 CPUs, 64.0 GB RAM, 723.4/926.4 GB disk)

OS                     macOS-15.7.7-arm64-arm-64bit
Environment            Darwin
Python                 3.12.13
Install                pip
Path                   /Users/aiuser/Projects/yolov26/.venv/lib/python3.12/site-packages/ultralytics
RAM                    64.00 GB
Disk                   723.4/926.4 GB
CPU                    Apple M2 Max
CPU count              12
GPU                    None
GPU count              None
CUDA                   None

numpy                  ✅ 2.1.3>=1.23.0
matplotlib             ✅ 3.10.9>=3.3.0
opencv-python          ✅ 4.13.0.92>=4.6.0
pillow                 ✅ 12.2.0>=7.1.2
pyyaml                 ✅ 6.0.3>=5.3.1
requests               ✅ 2.34.2>=2.23.0
scipy                  ✅ 1.17.1>=1.4.1
torch                  ✅ 2.12.0>=1.8.0
torch                  ✅ 2.12.0!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision            ✅ 0.27.0>=0.9.0
psutil                 ✅ 7.2.2>=5.8.0
polars                 ✅ 1.40.1>=0.20.0
ultralytics-thop       ✅ 2.0.19>=2.0.18
{'OS': 'macOS-15.7.7-arm64-arm-64bit', 'Environment': 'Darwin', 'Python': '3.12.13', 'Install': 'pip', 'Path': '/Users/aiuser/Projects/yolov26/.venv/lib/python3.12/site-packages/ultralytics', 'RAM': '64.00 GB', 'Disk': '723.4/926.4 GB', 'CPU': 'Apple M2 Max', 'CPU count': 12, 'GPU': None, 'GPU count': None, 'CUDA': None, 'Package Info': {'numpy': '✅ 2.1.3>=1.23.0', 'matplotlib': '✅ 3.10.9>=3.3.0', 'opencv-python': '✅ 4.13.0.92>=4.6.0', 'pillow': '✅ 12.2.0>=7.1.2', 'pyyaml': '✅ 6.0.3>=5.3.1', 'requests': '✅ 2.34.2>=2.23.0', 'scipy': '✅ 1.17.1>=1.4.1', 'torch': '✅ 2.12.0!=2.4.0,>=1.8.0; sys_platform == "win32"', 'torchvision': '✅ 0.27.0>=0.9.0', 'psutil': '✅ 7.2.2>=5.8.0', 'polars': '✅ 1.40.1>=0.20.0', 'ultralytics-thop': '✅ 2.0.19>=2.0.18'}}

### Minimal Reproducible Example

```
from ultralytics import YOLO

model = YOLO("store/models/yolo26m.pt")

results = model.export(
    format="tflite",
    imgsz=640,
    int8=True,
    data="visdrone-2019-det-optimal-split_small/dataset.yaml",
    verbose=True,
    fraction=0.30
)
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @kjmiller944, thank you for reporting this issue with Ultralytics 🚀. This is an automated response to help gather the right debugging details, and an Ultralytics engineer will also assist you soon.

Thanks for already including a **minimum reproducible example** and detailed environment information 🙌. Since this appears to be a 🐛 Bug Report related to `model.export()` with YOLO26 TFLite INT8 export, please also share any additional full export console logs if available so the team can better trace where the process is stalling. If needed, here is our guide for a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/).

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips).

We also recommend a visit to the [Docs](https://docs.ultralytics.com/) for usage references, including [Python](https://docs.ultralytics.com/usage/python) and [CLI](https://docs.ultralytics.com/usage/cli) examples.

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Please upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify this issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes) and [Tasks](https://docs.ultralytics.com/tasks) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by kjmiller944** [comment]
I am using the latest version of ultralytics.

**Comment by banu4prasad** [comment]
```
pip install ultralytics[export]
```

```python
import os
# Force single-threaded execution to prevent macOS threading deadlocks
os.environ["OMP_NUM_THREADS"] = "1"
os.environ["TF_NUM_INTRAOP_THREADS"] = "1"
os.environ["TF_NUM_INTEROP_THREADS"] = "1"

from ultralytics import YOLO

model = YOLO("store/models/yolo26m.pt")

exported_path = model.export(
    format="tflite",
    imgsz=640,
    int8=True,
    data="visdrone-2019-det-optimal-split_small/dataset.yaml",
    verbose=True,
    fraction=0.30
)
```

**Comment by glenn-jocher** [comment]
Thanks for sharing this; it looks like the stall is happening in the TensorFlow/`onnx2tf` conversion stage on macOS rather than in `model.export()` itself, so @kjmiller944 please try the `ultralytics[export]` install above in a fresh environment, restart the Python session afterward, and if it still hangs paste the full text export log plus `ultralytics.utils.checks.collect_system_info()` from that same environment so we can narrow down the exact dependency combination against the [TFLite export guide](https://docs.ultralytics.com/integrations/tflite).

**Comment by smqd19** [comment]
I've encountered similar issues with model quantization, especially with TensorFlow Lite and Apple Silicon. Here are a few things to check:

- **Compatibility**: Ensure that the `tflite` format is fully supported for your model architecture and that there are no known issues with Apple M2 Max. You might want to try exporting to another format like ONNX or TorchScript to see if the issue persists.
- **Verbose Logging**: Increase the verbosity of the logging to get more insights into where it's hanging. Sometimes, the logs can provide clues about what's causing the delay.
- **Resource Constraints**: Although you have ample RAM, the issue might be related to disk I/O or CPU throttling. Try running the export on a different machine or with a smaller dataset to see if it completes.

Additionally, consider updating to the latest version of Ultralytics and TensorFlow Lite, as there might be bug fixes or improvements that address this issue. If the problem continues, it might be worth reaching out to the Ultralytics team or checking their documentation for any known issues with quantization on Apple Silicon.

**Comment by glenn-jocher** [comment]
Thanks — one concrete way to isolate this is that TFLite export first builds a TensorFlow `_saved_model`, so please try:

```python
model.export(format="saved_model", int8=True, data="visdrone-2019-det-optimal-split_small/dataset.yaml", fraction=0.30)
model.export(format="tflite", int8=False)
```

If the `saved_model` run also stalls, that points to the TensorFlow conversion path on Apple Silicon rather than INT8 calibration itself; please paste the full text logs from both runs plus `ultralytics.utils.checks.collect_system_info()` from that same session, and we’ll compare against the supported [TFLite export settings](https://docs.ultralytics.com/integrations/tflite).

**Comment by raimbekovm** [comment]
Hi @kjmiller944, on macOS Apple Silicon `onnx>=1.18` currently triggers a stall in the TFLite INT8 conversion path. The pin in `pip install "ultralytics[export]"` enforces `onnx<1.18` for this reason, but a plain `pip install ultralytics` does not.

Could you try this in a fresh Python session, then re-run your export:

```bash
pip install "onnx<1.18.0"
```

Please share whether the INT8 TFLite output completes for you.
